tests/devices/test_default_qutrit.py

```diff
@@ -106,6 +106,24 @@ def test_apply_operation_single_wire_no_parameters(
         assert np.allclose(qutrit_device_1_wire._state, np.array(expected_output), atol=tol, rtol=0)
         assert qutrit_device_1_wire._state.dtype == qutrit_device_1_wire.C_DTYPE
 
+    @pytest.mark.parametrize("operation, expected_output, input, subspace", test_data_no_parameters)
+    def test_apply_operation_single_wire_no_parameters_adjoint(
+        self, qutrit_device_1_wire, tol, operation, input, expected_output, subspace
+    ):
+        """Tests that applying an adjoint operation yields the expected output state for single wire
+        operations that have no parameters."""
+        qutrit_device_1_wire._state = np.array(input, dtype=qutrit_device_1_wire.C_DTYPE)
+        qutrit_device_1_wire.apply(
+            [
+                qml.adjoint(operation(wires=[0]))
+                if subspace is None
+                else qml.adjoint(operation(wires=[0], subspace=subspace))
+            ]
+        )
+
+        assert np.allclose(qutrit_device_1_wire._state, np.array(expected_output), atol=tol, rtol=0)
+        assert qutrit_device_1_wire._state.dtype == qutrit_device_1_wire.C_DTYPE
```

Comment:
I'm confused - it looks like the inputs and assertions for this test are identical to the non-adjoint version, and they only differ in whether the gates are applied directly or the adjoint gates are applied. Is that correct, or am I missing something? And if correct, is it expected?

Reply:
The input and expected output are switched. See the `@pytest.mark.parametrize` statement in the test and on its non-adjoint equivalent.